File: src/main/java/com/fixtab/app/models/db/employees/EmployeeModel.java
package com.fixtab.app.models.db.employees;

import com.fixtab.app.models.db.BaseEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.Date;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Table(name = "employees")
@Entity
public class EmployeeModel extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employeeid")
    private int employeeId;

    @Column(name = "name")
    private String name;

    @Column(name = "surname")
    private String surname;

    @Column(name = "phonenumber")
    private String phoneNumber;

    @Column(name = "email")
    private String email;

    @Column(name = "birthdate")
    private Date birthDate;

    @Column(name = "gender")
    private String gender;

    @Column(name = "pesel")
    private String pesel;

    @Column(name = "targetobjectid")
    private Integer targetObjectId;

    @Column(name = "roleid")
    private Integer roleId;
}

File: client-application/src/app/entitites/employee-model.ts
interface IEmployee {
  employeeId?: number;
  name?: string;
  surname?: string;
  email?: string;
  phoneNumber?: string;
  birthDate?: string;
  gender?: string;
  pesel?: string;
  addressId?: number;
  targetObjectId?: number;
  roleId?: number;
}

interface IEmployeePassword {
  oldPassword?: string;
  password?: string;
}

export class Employee implements IEmployee {
  constructor(
   public employeeId?: number,
   public name?: string,
   public surname?: string,
   public email?: string,
   public birthDate?: string,
   public phoneNumber?: string,
   public gender?: string,
   public pesel?: string,
   public addressId?: number,
   public targetObjectId?: number,
   public roleId?: number
  ) {}
}

export class EmployeePassword implements IEmployeePassword {
  constructor(
    public oldPassword?: string,
    public password?: string
  ) {}
}
